core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,290 @@
+            for (String key : apiResponses.keySet()) {
+                if (apiResponses.get(key).getContent() != null && (isGreenStatus(key) || key.equals("default"))) {
+                    return apiResponses.get(key).getContent().keySet();
+                }
+            }
+        }
+        return new HashSet<>();
+    }
+
+    private  Component createComponent(HashMap<String, String> partNames, Operation operation,PathItem.HttpMethod operationName, String pathName) {
+        String componentName = buildName(operation.getOperationId(), partNames);
+        String componentType = "http.request";
+        String componentDescription = operation.getDescription();
+        ComponentVersion componentVersion = new ComponentVersion(new ComponentVersionKey(componentName, versionNumber),componentDescription);
+
+        List<ComponentParameter> infos = getInfos(componentName,pathName, operationName);
+        List<ComponentParameter> queryParams = getQueryParams(componentName, operation.getParameters());
+        List<ComponentParameter> headers = getHeaders(componentName, partNames);
+        List<ComponentParameter> params = Stream.of(infos, queryParams, headers).flatMap(Collection::stream).collect(Collectors.toList());
+        Component component = new Component(new ComponentKey(componentName, versionNumber), componentType, componentName, componentDescription, componentVersion, params, new ArrayList<>());
+        return component;
+    }
+
+    private  List<ComponentParameter> getInfos(String componentName, String pathName, PathItem.HttpMethod operationName) {
+        return new ArrayList(){{
+            add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, "endpoint"), pathName));
+            add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, "type"), operationName.name()));
+            add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, "connection"), connectionName));
+        }};
+    }
+
+    private  List<ComponentParameter> getQueryParams(String componentName, List<Parameter> parameters) {
+        Integer counter = 1;
+        List<ComponentParameter> queryParams = new ArrayList();
+
+
+        if (parameters != null) {
+            for (int i = 0; i < parameters.size(); i++ ) {
+                Parameter parameter = parameters.get(i);
+                if (parameter.getIn() != null && parameter.getIn().equals("query")) {
+                    String parameterName = parameter.getName();
+                    queryParams.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber,String.format("queryParam.%s", counter) ), String.format("%s, #%s#", parameterName, parameterName)));
+                    counter += 1;
+                }
+            }
+        }
+        return queryParams;
+    }
+
+
+    private  List<ComponentParameter> getHeaders(String componentName, HashMap<String, String> partNames) {
```

Comment:
there could be other headers defined in the OpenApi documentation other than the security, request and response headers currently stored in the partNames